### PR TITLE
Workflow backend actor factory.

### PR DESCRIPTION
**Pull request by francares**
This is not the complete code. I'm waiting for dependencies: ShadowWorkflowActor and BackendConfiguration impl. to finish with it.

### CODE REVIEW

File: backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    * @param backendWorkflowDescriptor      Workflow level data needed to be able to execute a workflow.
+    * @param backendConfigurationDescriptor Backend configuration data.
+    * @param serviceRegistryActor           Cromwell service registry.
+    * @return An actor ref pointing to a WorkflowBackendActor implementation.
+    */
+  def getBackend(initClass: String,
+                 actorSystem: ActorSystem,
+                 backendWorkflowDescriptor: BackendWorkflowDescriptor,
+                 backendConfigurationDescriptor: BackendConfigurationDescriptor,
+                 serviceRegistryActor: ActorRef): ActorRef
```
Comment (cjllanwarne) [inline backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala]: Also the list of calls which this Backend will be asked to execute

Comment (francares) [inline backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala]: Right! I will add it.

**Comment by francares** [comment]
I would like to merge this interface so I can continue with other task until all dependencies are satisfied.

@cjllanwarne and @scottfrazer Can you review this small piece of code?

**Comment by cjllanwarne** [comment]
:+1:

File: backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    * @param serviceRegistryActor           Cromwell service registry.
+    * @param calls                          A sequence of WDL Calls used to validate runtime requirements by the backend.
+    * @return An actor ref pointing to a WorkflowBackendActor implementation.
+    */
+  def getBackend(initClass: String,
+                 actorSystem: ActorSystem,
+                 backendWorkflowDescriptor: BackendWorkflowDescriptor,
+                 backendConfigurationDescriptor: BackendConfigurationDescriptor,
+                 serviceRegistryActor: ActorRef,
+                 calls: Seq[Call]): ActorRef
```
Comment (scottfrazer) [inline backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala]: I don't think we need to pass in `calls`... We can just do `backendWorkflowDescriptor.workflowNamespace.workflow.calls`

Comment (scottfrazer) [inline backend/src/main/scala/cromwell/backend/WorkflowBackendActorFactory.scala]: Oh wait, I think I'm wrong, forget that... these are a subset of the calls that will run on that backend.

**Comment by scottfrazer** [comment]
:+1:

**Pull request merged**